client/state/sites/reducer.js

```diff
@@ -135,43 +135,58 @@ export function items( state = {}, action ) {
 		case SITE_SETTINGS_UPDATE:
 		case SITE_SETTINGS_RECEIVE: {
 			const { siteId, settings } = action;
+			const site = state[ siteId ];
 
-			// A site settings update may or may not include the icon property.
-			// If not, we should simply return state unchanged.
-			if ( ! settings.hasOwnProperty( 'site_icon' ) ) {
+			if ( ! site ) {
 				return state;
 			}
 
-			const mediaId = settings.site_icon;
+			let nextSite = { ...site };
 
-			// Similarly, return unchanged if next icon matches current value,
-			// accounting for the fact that a non-existent icon property is
-			// equivalent to setting the media icon as null
-			const site = state[ siteId ];
-			if ( ! site || ( ! site.icon && null === mediaId ) ||
-					( site.icon && site.icon.media_id === mediaId ) ) {
-				return state;
-			}
+			return reduce( [ 'blog_public', 'site_icon' ], ( memo, key ) => {
+				// A site settings update may or may not include the icon or blog_public property.
+				// If not, we should simply return state unchanged.
+				if ( ! settings.hasOwnProperty( key ) ) {
+					return memo;
+				}
 
-			let nextSite;
-			if ( null === mediaId ) {
-				// Unset icon
-				nextSite = omit( site, 'icon' );
-			} else {
-				// Update icon, intentionally removing reference to the URL,
-				// shifting burden of URL lookup to selector
-				nextSite = {
-					...site,
-					icon: {
-						media_id: mediaId
-					}
-				};
-			}
+				switch ( key ) {
+					case 'blog_public':
+						const isPrivate = parseInt( settings.blog_public, 10 ) !== 1;
+						nextSite.is_private = isPrivate;
+						break;
 
-			return {
-				...state,
-				[ siteId ]: nextSite
-			};
+					case 'site_icon':
+						const mediaId = settings.site_icon;
+						// Return unchanged if next icon matches current value,
+						// accounting for the fact that a non-existent icon property is
+						// equivalent to setting the media icon as null
+						if ( ( ! site.icon && null === mediaId ) ||
+								( site.icon && site.icon.media_id === mediaId ) ) {
+							return memo;
+						}
+
+						if ( null === mediaId ) {
+							// Unset icon
+							nextSite = omit( nextSite, 'icon' );
+							break;
```

Comment:
This `break;` is redundant since there's no other logic in the `case` to be run except the final `break;` anyways.